common/metrics/tags.go

```diff
@@ -58,14 +62,23 @@ type (
 	}
 )
 
+// sanitizer is used to sanitize tag values, to prevent collisions and invalid data.
+// values should not allow dots for "namespacing" / "grouping" metrics, as they seem to refer to "one thing" in all cases.
+var sanitizer = tally.NewSanitizer(tally.SanitizeOptions{
+	NameCharacters:       tally.ValidCharacters{tally.AlphanumericRange, tally.UnderscoreDashCharacters},
```

Comment:
Just curious.
AFAIK, Prometheus is not supporting dashes: https://prometheus.io/docs/concepts/data_model/#metric-names-and-labels (`It must match the regex [a-zA-Z_:][a-zA-Z0-9_:]*.`)

If user selects Prometheus  as a reporter, will dashes be replaced to underscore?

Reply:
good question... hopefully?  that'd be a good reason to not add dashes though, if not

Reply:
let me see if I can find any other stat-sanitizing code that might be happening in this repo.  it may reveal more details.

Reply:
Can we leave the dash out and merge this PR?

Reply:
yeah, I'd be game to do that.  I haven't yet looked at other stats, but "the value is one conceptual thing" seems desirable from all calls to these at least.

Reply:
updated, dropped `-` (and added to the comment).  so this is now:
```go
// sanitizer is used to sanitize tag values, to prevent collisions and invalid data.
// values should not allow dots for "namespacing" / "grouping" metrics, as they seem to refer to "one thing" in all cases.
// `-` is not allowed in Prometheus keys or names, but it may be reasonable to allow it in values.
var sanitizer = tally.NewSanitizer(tally.SanitizeOptions{
	NameCharacters:       tally.ValidCharacters{tally.AlphanumericRange, tally.UnderscoreCharacters},
	KeyCharacters:        tally.ValidCharacters{tally.AlphanumericRange, tally.UnderscoreCharacters},
	ValueCharacters:      tally.ValidCharacters{tally.AlphanumericRange, tally.UnderscoreCharacters},
	ReplacementCharacter: '_',
})
```